### ISSUE TITLE
Incorrect behavior when running .py files in the latest version

### ISSUE THREAD

**Issue by kuri54** [microsoft/vscode-jupyter#7369]
I updated VSCode since my research activities have come to an end. At that time, I also updated the extensions, but now I can't do what I used to be able to do.

## Expected behaviour
extension versions
Pyhon: 2021.8.1105858891
Jupyter: 2021.8.1054968649

1. Open the `.py` file
1. Create a code indent with `# %%` and execute with `shift + Enter`
1. The Interactive window appears and the code is executed
1. Close the Interactive window
1. Same behavior as 2.
1. The code will be executed again.

![old](https://user-images.githubusercontent.com/40049003/131606206-d774e14a-047c-4a49-9971-3710a40ef337.gif)

## Actual behavior
Extension versions
Pyhon: 2021.8.1159798656
Jupyter: 2021.8.1236758218

* The code is not executed again.
* The preview window appears instead of the Interactive window.
* The message `Direct kernel connection broken` is displayed.

![latest](https://user-images.githubusercontent.com/40049003/131606259-e8209137-5a04-4649-96ce-4b05ac004256.gif)

The icon displayed in the Interactive window seems to have been changed.
Latest
![スクリーンショット 2021-09-01 11 24 38](https://user-images.githubusercontent.com/40049003/131606313-7c591af9-26b5-4864-9f9d-ccf79e002dd1.jpg)

Old
![スクリーンショット 2021-09-01 11 27 20](https://user-images.githubusercontent.com/40049003/131606327-e05ea4e2-af49-400e-8348-c26e94e4329f.jpg)

## Logs
Jupyter
```
Error 2021-09-01 02:49:26: DataScience Error [Error: Direct kernel connection broken
    at E.getDisposedError (/home/kurita_10/.vscode-server/extensions/ms-toolsai.jupyter-2021.8.1236758218/out/client/extension.js:37:840597)
    at f.P.handleCodeRequest (/home/kurita_10/.vscode-server/extensions/ms-toolsai.jupyter-2021.8.1236758218/out/client/extension.js:90:205970)
    at /home/kurita_10/.vscode-server/extensions/ms-toolsai.jupyter-2021.8.1236758218/out/client/extension.js:90:220305]
```

## Steps to Reproduce
1. Make sure your extensions (Python, Jupyter) are up to date.
1. Start Docker from a dockerfile with a port output to JupyterNotebook.
1. Connect from VSCode's remote feature.
1. Open the .py file and run the code.
(Do you need to provide a dockerfile?)

I don't know which extension has the problem this time, Python or Jupyter. (Please forgive me for posting on this issue without knowing.)  
I tried using an older environment for either version, but that didn't solve the problem.  
I don't think the Docker connection is the problem, what do you think?

**Comment by rchiodo** [comment]
This sounds like a timing issue with creation of the kernel in the new interactive window. 

Can you include your entire 'jupyter' output log? It shows up here:

![image](https://user-images.githubusercontent.com/19672699/131705455-63b1e80b-24b8-45dd-af03-863c10bf1f38.png)

I'm going to enter a separate issue for the image next to the code. That's basically the execution count missing when code is collapsed.

**Comment by kuri54** [comment]
Thanks for replying!  
<br />

> This sounds like a timing issue with the creation of the kernel in the new interactive window.

Yes, it does. This sounds like a timing issue with creation of the kernel in the new interactive window.  
<br />

> It shows up here:

I tried to respond to this reply, but for some reason I can't update to the latest version. I was able to do it yesterday, is there something wrong with it?
<br />

However, the log shown is as described above. I don't recall seeing any other error messages at the top of that log other than this one.I am sorry that I cannot reproduce this.
<br />

> I'm going to enter a separate issue for the image next to the code. That's basically the execution count missing when code is collapsed.

Thanks for setting up a new issue.
